### ISSUE TITLE
spideroak-7.0.1 no longer builds

### ISSUE THREAD

**Issue by goertzenator** [NixOS/nixpkgs#37629]
## Issue description

Installing spideroak-7.0.1 fails with a sha256 error.

The problem is that the spideroak website now has a link that downloads only the current version which recently got bumped to 7.1.0.  Prior to the 7.0.1, you could encode the version in the url to get a specific version.

Current URL (can't tell what version you get)
```
https://spideroak.com/release/spideroak/slack_tar_x86_64
```

Old URL (can pick version, but no longer works for 7.x.y)
```
https://spideroak.com/getbuild?platform=slackware&arch=x64&version=6.1.9
```

I'm going to poke the spideroak people about this issue and report what I find.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.86, NixOS, 17.09.3194.edf1079d43f (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre131327.0e7c9b32817, unstable-18.03pre130569.7a04c2ca296"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

Tagging @neilmayhew and @lloydhazlett since they've recently worked on this.

**Comment by lloydhazlett** [comment]
Well, that didn't take long :disappointed:

I did ask SpiderOak to make versioned downloads available and received a reply 11th March:

>  I'm sorry to report that a new build process that we have implemented puts former builds on our internal network and are not publically available. We did consider the difficulty that this will cause package builders, and looked for a way to accommodate that, but unfortunately we were not able to come up with a way to do that within our new build process.
>
> We can provide you with any particular prior version you need, but I recognize that is not what you and other package builders need. Sorry for the bad news. As a Linux user myself I wish there were something I could do to help.
>
> You didn't ask, but I will mention that any version prior to our current 7.0.1 does not ship with our current cert and for that reason will no longer be fully functional. So at least for today everyone ought to be building and running our current 7.0.1. That doesn't address your long term need however.

Having looked at what approach the [Arch spideroak package](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=spideroak-one) takes, it seems they have the same problem.

I can submit a PR to update the hashes and bump the version, but I'm not sure if there's a better way to address this long term if SpiderOak won't resume publishing versioned downloads.

**Comment by neilmayhew** [comment]
I did notice this:

```
$ wget -S --spider https://spideroak.com/release/spideroak/slack_tar_x64
...
  Content-Disposition: attachment; filename=SpiderOakONE-7.1.0-slack_tar_x64.tgz
```

So at least it's possible to figure out what version is being returned. It's also possible to supply GET parameters at the end of the URL that are ignored, so it's possible to have unique URLs for different versions. However, it's not possible to stop the content of those URLs from changing their hashes, and it's not possible to retrieve previous versions once a new one has been released.

**Comment by goertzenator** [comment]
Spideroak support got back to me and pointed me to https://spideroak.com/browse/share/golden/retriever

7.0.1 is there, but 7.1.0 is not.  I was able to grab 7.0.1 with curl.  It was slow and the url is messy, but this at least offers a near term fix for 7.0.1.

They also suggested the old url scheme, but I couldn't make it work.  I suspect the support individual was not aware that the build process had changed.

BTW it looks like Gentoo will also be affected, but I can't see that anyone has complained about it yet.
